#!/usr/bin/env python
"""Configuration options for the scene graph visualization."""
PACKAGE = "hydra_utils"

import dynamic_reconfigure.parameter_generator_catkin as dr_gen  # NOQA

gen = dr_gen.ParameterGenerator()

gen.add(
    "layer_z_step",
    dr_gen.double_t,
    0,
    "size of the distance between layers",
    5.0,
    0.0,
    50.0,
)
gen.add(
    "mesh_edge_break_ratio",
    dr_gen.double_t,
    0,
    "point at which to break the edge into many edges",
    0.5,
    0.0,
    1.0,
)
gen.add(
    "mesh_layer_offset",
    dr_gen.double_t,
    0,
    "height mesh layer exists at",
    0.0,
    -5.0,
    5.0,
)
gen.add(
    "collapse_layers",
    dr_gen.bool_t,
    0,
    "whether or not to show the layers independently",
    False,
)
gen.add(
    "color_places_by_distance",
    dr_gen.bool_t,
    0,
    "color the places by distance to nearest obstacle",
    False,
)

gen.add(
    "places_colormap_min_distance", dr_gen.double_t, 0, "min distance", 0.0, 0.0, 100.0
)
gen.add(
    "places_colormap_max_distance", dr_gen.double_t, 0, "max distance", 10.0, 0.0, 100.0
)

exit(gen.generate(PACKAGE, PACKAGE, "Visualizer"))
